File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.js
/**
 * @param {number} n
 * @param {number} t
 * @return {number}
 */
var smallestNumber = function(n, t) {
    let current = n;
    while(true){
        let digProduct = 1;
        const strNum = current.toString();
        
        for (let i = 0; i < strNum.length; i++) {
            digProduct *= parseInt(strNum[i], 10);
        }
        // Check if the product is divisible by t
        if (digProduct % t === 0) {
            return current;
        }
        current++;
    }
};

File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {
        int current = n;
        
        while (true) {
            int digitProduct = 1;
            String strNum = String.valueOf(current);
            
            for (int i = 0; i < strNum.length(); i++) {
                digitProduct *= strNum.charAt(i) - '0';
            }
            // Check if the product is divisible by t
            if (digitProduct % t == 0) {
                return current;
            }
            current++;
    }
}}